File: coding-interview/js/array/c2.js
// Single number
function findSingleNum(numbers) {
  const length = numbers.length;
  let result = numbers[0];

  for (let i = 1; i < length; i++) {
    result = result ^ numbers[i];
  }

  return result;
}

const numbers = [4, 1, 2, 1, 2];

console.log(findSingleNum(numbers));

File: pattern/2.PatternTwoPointers/c2.py
# Remove Duplicates (easy)

# def rm_duplicates(numbers):
#     length = len(numbers)
#     if not length:
#         return -1

#     result = []
#     result.append(numbers[0])
#     idx = 1

#     while (idx < length):
#         element = numbers[idx]
#         if element != result[len(result) - 1]:
#             result.append(element)
#         idx += 1

#     return len(result)

def rm_duplicates(numbers):
    length = len(numbers)
    if not length:
        return -1

    result = 1
    idx = 1

    while (idx < length):
        element = numbers[idx]
        if element != numbers[idx - 1]:
            result += 1
        idx += 1

    return result


numbers = [2, 3, 3, 3, 6, 9, 9]

print(rm_duplicates(numbers))
print(rm_duplicates([2, 2, 2, 11]))
